internal/services/redisenterprise/redis_enterprise_cluster_resource.go

```diff
@@ -103,7 +103,7 @@ func resourceRedisEnterpriseCluster() *pluginsdk.Resource {
 				Deprecated: "This field currently is not yet being returned from the service API, please see https://github.com/Azure/azure-sdk-for-go/issues/14420 for more information",
 			},
 
-			"tags": tags.SchemaDataSource(),
+			"tags": tags.Schema(),
```

Comment:
actually this resource doesn't support update, so this has to be ForceNew unless we implement an Update function here:

>         	* resource azurerm_redis_enterprise_cluster: No Update defined, must set ForceNew on: []string{"tags"}
366

Reply:
So the original PR should be reverted - right now, the resource is just broken as no tags can be set - I am happy to close this PR out in favour of moving back to the ForceNew - happy for you to make the call here

Reply:
imho if the API supports update, we should implement an update function and correctly implement tags - if it doesn't then we have to use forcenew

Reply:
I'm working on that right now, but I'm doing that to teach myself go. If anyone wants to go ahead they are welcome to!

Reply:
I'm doing that here: https://github.com/hashicorp/terraform-provider-azurerm/pull/13084